src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -140,7 +140,8 @@ class IOUAmountPage extends React.Component {
      */
     validateAmount(amount) {
         const decimalNumberRegex = new RegExp(/^\d+(,\d+)*(\.\d{0,2})?$/, 'i');
-        return amount === '' || (decimalNumberRegex.test(amount) && (parseFloat((amount * 100).toFixed(2)).toString().length <= CONST.IOU.AMOUNT_MAX_LENGTH));
+        const lengthBeforeDecimal = this.stripCommaFromAmount(amount.split('.')[0]).length;
+        return amount === '' || (decimalNumberRegex.test(amount) && decimalNumberRegex.test(amount) && lengthBeforeDecimal <= (CONST.IOU.AMOUNT_MAX_LENGTH);
```

Comment:
Why do need to check the decimal part only? It seems to be that we just need to strip `,` and `.` from the number. But in reality, the value will never have the `,`.

Reply:
Because the regex enforces the number of values after the decimal to `<= 2.` If the condition is imposed on the whole string instead, user can have `500000` and `5000.0` in his valid range but not `5000.24` if  ` IOU.AMOUNT_MAX_LENGTH =6`. Applying check on only part before decimal will ensure a uniform range for the user. 

Also commas were kept because they were expected to be handled in the existing code. I will happily remove them if needed (We might be inserting commas if we are copy-pasting the amount)!

Reply:
Yeah, commas are added. Missed that upper line.  On Backend, we check the whole number via converting the floats to integer  `(amount * 100)` is used. 
If we try the logic from the previous code. 
1. 5000.24 => 500024
2. 5000.0  => 500000

I will try to stick to the old code. and we can just prevent zeros.

Let me know if I am missing something.

Reply:
The logic that was previously provided works well for a processed string (as we get in backend ) since we can be happy without caring about zeroes in the beginning. In this case, we will need to add an additional condition that essentially checks if (length of stripped portion as a parsedFloat + length of trailing zeroes ) doesn't exceed the limit.

**Previous Code:**
00500.23  => disregards the initial two zeroes, applies the older condition, and checks from the amount string if the number of zeroes, in the beginning, invalidates the amount string.

**Newer Logic**
00500.23 => takes care of the length of the string directly. The two conditions; (1) fractional part to 2 decimal places (2) Length of string before decimal should be bounded by Amount_Length.

In my humble opinion, this condition will ensure a lesser overhead, cleaner, and more generally applicable too! Do let me know your thoughts on it :D .

Reply:
> disregards the initial two zeroes, applies the older condition, and checks from the amount string if the number of zeroes in the beginning, invalidates the amount string.

Hmm. Still, I am not sure of what you explained above but I got the idea.

Reply:
Since we essentially cast strings as integers, we lose the zeroes in the beginning of the number that causes the referenced issue ( Since parseFloat("00000005")=5 ).
I am suggesting that if we directly use the string for validation, we won't be losing the zeroes in the beginning.